### PR TITLE
Switch from Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by pasitlaothamatas**
### Summary

- Decided to switch from Travis CI to Github Actions after Travis CI's new pricing model
- Unlimited build minutes and unlimited parallel jobs on Github Actions

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: .github/workflows/manual.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# This is a basic workflow that is manually triggered
+
+name: UBC Formula Electric CI
+on: [pull_request, push]
```
Comment (pasitlaothamatas) [inline .github/workflows/manual.yml:4]:
```suggestion
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
```

**Comment by garethellis0** [comment]
Seems legit to me, I'd just make sure that all the possible failure cases actually fail CI

**Comment by pasitlaothamatas** [comment]
@garethellis0 gotcha. Also will try to get series ci working. Might just run both travis ci and github actions for now and when series ci is working we can drop travis ci.

File: scripts/travis_ci/travis_script.sh [hunk 2]
```diff
@@ -23,13 +23,17 @@ if [ "$RUN_ARM_BUILD" = "true" ]; then
...
-        else
+        #    SHA=${TRAVIS_COMMIT}
+        #else
             # Travis CI Pull Request build
-            SHA=${TRAVIS_PULL_REQUEST_SHA}
-        fi
+        #    SHA=${TRAVIS_PULL_REQUEST_SHA}
+        #fi
+        #SHA=${GITHUB_SHA}
+        SHA=${GITHUB_REF}
```
Comment (pasitlaothamatas) [inline scripts/travis_ci/travis_script.sh:34]:
```suggestion
        SHA=${GITHUB_SHA}
```

File: .github/workflows/githubactions.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                                - Name: "Check STM32CubeMX Code Generation"
+                                  env: RUN_CUBE_CODEGEN_CHECKS 
+                env: 
+                        INSTALL_DIR: /usr/local
+                        CUBE_INSTALL_DIR: $INSTALL_DIR/STM32CubeMX
+
+                steps:               
+                        - name: Install Addons
+                          run: |               
+                                  echo ${{ github.event.push.head.sha }}
```
Comment (pasitlaothamatas) [inline .github/workflows/githubactions.yml:32]:
```suggestion
                                  echo $GITHUB_SHA
```

File: scripts/travis_ci/travis_script.sh [hunk 4]
```diff
@@ -23,13 +23,14 @@ if [ "$RUN_ARM_BUILD" = "true" ]; then
...
             # Travis CI Branch build
-            SHA=${TRAVIS_COMMIT}
-        else
+        #    SHA=${TRAVIS_COMMIT}
+        #else
             # Travis CI Pull Request build
-            SHA=${TRAVIS_PULL_REQUEST_SHA}
-        fi
+        #    SHA=${TRAVIS_PULL_REQUEST_SHA}
+        #fi
```
Comment (pasitlaothamatas) [inline scripts/travis_ci/travis_script.sh:32]: Change SHA through .yml instead

File: .github/workflows/githubactions.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+                steps:                                      
+                        - name: Install Addons
+                          run: |               
+                                  sudo apt-get update
+                                  sudo dpkg --add-architecture i386
+                                  sudo apt-get install libncurses5 libbz2-1.0 libz1 valgrind
+                                                                                                                                  
+                        - uses: actions/checkout@v2
+                        - name: Install-Python-Packages
```
Comment (pasitlaothamatas) [inline .github/workflows/githubactions.yml:35]:
```suggestion
                        - name: Install Python
```

**Comment by pasitlaothamatas** [comment]
@garethellis0 Series CI is now working so ready for a final pass review

**Comment by kirons96** [comment]
https://images.newschoolers.com/images/17/00/44/22/01/442201_1600w.png

**Comment by garethellis0** [comment]
> @garethellis0 gotcha. Also will try to get series ci working. Might just run both travis ci and github actions for now and when series ci is working through github actions we can drop travis ci.

Sorry, just seeing this now. This is a good way to do it. :+1:

**Review by garethellis0 — APPROVED** [review]

**Pull request merged**